CMakeLists.txt

```diff
@@ -26,6 +26,34 @@ target_compile_features(fplus INTERFACE cxx_std_14)
 find_package(Threads REQUIRED)
 target_link_libraries(fplus INTERFACE Threads::Threads)
 
+# ---- Create auto-generated fwd and curried functions ----
+
+if(UNIX)
+  add_custom_target(
+      generate_derived_functions
+      ALL
+      COMMAND bash "${PROJECT_SOURCE_DIR}/generate/generate_derived_functions.sh"
+      BYPRODUCTS "${PROJECT_SOURCE_DIR}/include/fplus/fwd_instances.autogenerated_defines" "${PROJECT_SOURCE_DIR}/include/fplus/curry_instances.autogenerated_defines"
```

Comment:
To keep things on the left nice and tidy within 80 columns, `BYPRODUCTS` could go on its own line, followed by the byproducts on each new line.